src/docker/shared.rs

```diff
@@ -488,6 +489,284 @@ impl Directories {
     }
 }
 
+// the mount directory for the data volume.
+pub const MOUNT_PREFIX: &str = "/cross";
+// the prefix used when naming volumes
+pub const VOLUME_PREFIX: &str = "cross-";
+// default timeout to stop a container (in seconds)
+pub const DEFAULT_TIMEOUT: u32 = 2;
+// instant kill in case of a non-graceful exit
+pub const NO_TIMEOUT: u32 = 0;
+
+pub(crate) static mut CONTAINER: Container = Container::new();
+
+// we need to specify drops for the containers, but we
+// also need to ensure the drops are called on a
+// termination handler. we use an atomic bool to ensure
+// that the drop only gets called once, even if we have
+// the signal handle invoked multiple times or it fails.
+#[allow(missing_debug_implementations)]
+pub struct Container {
+    deleter: Option<DeleteContainer>,
+    exists: AtomicBool,
+}
+
+impl Container {
+    pub const fn new() -> Container {
+        Container {
+            deleter: None,
+            exists: AtomicBool::new(false),
+        }
+    }
+
+    pub fn create(engine: Engine, name: String) -> Result<()> {
+        // SAFETY: safe, since guarded by an atomic swap.
```

Comment:
safety comments should explain why things are safe, not that it is safe

```suggestion
        // SAFETY: guarded by an atomic swap
```

i know this is how it was previously but lets change it now :D